hazelcast/src/main/java/com/hazelcast/collection/CollectionContainer.java

```diff
@@ -114,15 +114,9 @@ protected void clearBackup(Set<Long> itemIdSet) {
     }
 
     protected boolean contains(Set<Data> valueSet) {
+        Collection<CollectionItem> collection = getCollection();
         for (Data value : valueSet) {
-            boolean contains = false;
-            for (CollectionItem item : getCollection()) {
-                if (value.equals(item.getValue())) {
-                    contains = true;
-                    break;
-                }
-            }
-            if (!contains) {
+            if (!collection.contains(new CollectionItem(-1, value))) {
```

Comment:
In order to prevent the cost caused by extra object creation in every iteration; what about an approach like this:
```
protected boolean contains(Set<Data> valueSet) {
        Collection<CollectionItem> collection = getCollection();
        CollectionItem item = new CollectionItem(-1, null); <--------------------- Create once.
        for (Data value : valueSet) {
            item.setValue(value);<----------------- Only set value
            if (!collection.contains(item)) {
                return false;
            }
        }
        return true;
    }
```